### PR TITLE
add docs about required system dependencies

### LINKED ISSUE #1775: Document that make and cmake are required to build
(due to jemalloc)

on ubuntu
`sudo apt install clang make cmake pkg-config libssl-dev`

### PR DESCRIPTION
**Pull request by trinity-1686a**
### Description

fix #1775
I added only packages I actually needed to run quickwit. I got no complain about make/cmake/pkg-config on neither platform so I assume they are probably no longer needed.

### How was this PR tested?

install the dependencies in `ubuntu:latest`, `amazonlinux:latest` and `archlinux:latest` and verify `cargo run --release` print quickwit help message

### CODE REVIEW

File: docs/get-started/installation.md [hunk 1]
```diff
@@ -61,6 +61,44 @@ pacman -S openssl
 
 </Tabs>
 
+Additionally it requires a few more dependencies to compile it. These dependencies are not required on production system:
```
Comment (ChillFish8) [inline docs/get-started/installation.md:64]:
```suggestion
Additionally, it requires a few more dependencies to compile it. These dependencies are not required on production systems running the produced binary:
```

**Comment by ChillFish8** [comment]
Can this just be checked if these are all that is needed when compiling with the `release-feature-set` feature enabled? This enables OpenSSL and the various other tools like Postgres and AWS that are disabled by default.

**Comment by trinity-1686a** [comment]
`pkg-config` (maybe more) is required with that feature. I'll update accordingly

**Review by ChillFish8 — APPROVED** [review]
LGTM

**Pull request merged**